### ISSUE TITLE
Tapping on the screen when running a test yields exception

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#10663]
## Steps to Reproduce

1. Run a flutter test via `flutter run`
2. Tap anywhere on the screen

## Expected behavior

Either nothing should happen, or we should do something useful

## Actual behavior

```
E/flutter (22199): [ERROR:../../lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter (22199): Bad state: No element
E/flutter (22199): #0      Iterable.lastWhere (dart:core/iterable.dart:549)
E/flutter (22199): #1      WidgetTester.dispatchEvent (package:flutter_test/src/widget_tester.dart:300)
E/flutter (22199): #2      LiveTestWidgetsFlutterBinding.dispatchEvent (package:flutter_test/src/binding.dart:838)
E/flutter (22199): #3      BindingBase&SchedulerBinding&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:86)
E/flutter (22199): #4      BindingBase&SchedulerBinding&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:45)
E/flutter (22199): #5      BindingBase&SchedulerBinding&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:38)
E/flutter (22199): #6      _dispatchPointerDataPacket (file:///b/build/slave/Linux_Engine/build/src/flutter/lib/ui/hooks.dart:55)
```

**Comment by tvolkert** [comment]
@Hixie

**Comment by TahaTesser** [comment]
Not able to reproduce this issue using default widget_test with counter template, when executing `flutter run -t test/widget_test.dart`

<details> 
<summary>logs</summary> 

``` 
Launching test/widget_test.dart on SM M305F in debug mode...
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                         8.2s
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...                 6.3s
Waiting for SM M305F to report its views...                         17ms
I/flutter (20744): 00:00 +0: Counter increments smoke test              
Syncing files to device SM M305F...                                421ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on SM M305F is available at: http://127.0.0.1:53962/Q-2IL2mFET4=/
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): Some possible finders for the widgets at Offset(446.3, 547.2):
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
I/flutter (20744): Some possible finders for the widgets at Offset(406.3, 309.8):
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
I/flutter (20744): Some possible finders for the widgets at Offset(393.0, 263.2):
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
I/flutter (20744): Some possible finders for the widgets at Offset(408.4, 270.0):
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
I/flutter (20744): Some possible finders for the widgets at Offset(403.3, 256.0):
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
I/flutter (20744): Some possible finders for the widgets at Offset(398.0, 244.2):
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
I/flutter (20744): Some possible finders for the widgets at Offset(400.6, 245.9):
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
I/flutter (20744): Some possible finders for the widgets at Offset(427.5, 261.1):
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
I/flutter (20744): Some possible finders for the widgets at Offset(411.7, 248.4):
I/flutter (20744):   find.byType(PrimaryScrollController)
I/flutter (20744):   find.byType(Scaffold)
I/flutter (20744):   find.byType(MyHomePage)
I/flutter (20744):   find.byType(FadeTransition)
I/flutter (20744):   find.byType(FractionalTranslation)
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
I/flutter (20744): No widgets found at Offset(158.6, 124.4).
I/flutter (20744): 00:03 +1: All tests passed!
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(325.0, 343.6).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(435.0, 374.5).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(279.1, 249.7).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(151.4, 161.7).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(111.1, 104.5).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(96.7, 17.8).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(352.9, 204.8).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): Some possible finders for the widgets at Offset(353.5, 306.4):
I/flutter (20744):   find.text('Test finished.')
I/flutter (20744):   find.byType(Center)
I/flutter (20744):   find.byType(Container)
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(420.7, 335.2).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(447.5, 353.0).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(116.4, 15.7).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(322.7, 251.0).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(427.7, 274.2).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(561.7, 282.3).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): No widgets found at Offset(647.1, 353.4).
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): Some possible finders for the widgets at Offset(393.6, 307.7):
I/flutter (20744):   find.text('Test finished.')
I/flutter (20744):   find.byType(Center)
I/flutter (20744):   find.byType(Container)
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): Some possible finders for the widgets at Offset(346.7, 310.6):
I/flutter (20744):   find.text('Test finished.')
I/flutter (20744):   find.byType(Center)
I/flutter (20744):   find.byType(Container)
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): Some possible finders for the widgets at Offset(323.2, 300.9):
I/flutter (20744):   find.text('Test finished.')
I/flutter (20744):   find.byType(Center)
I/flutter (20744):   find.byType(Container)
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 0
I/flutter (20744): Some possible finders for the widgets at Offset(281.3, 280.6):
I/flutter (20744):   find.text('Test finished.')
I/flutter (20744):   find.byType(Center)
I/flutter (20744):   find.byType(Container)
D/ViewRootImpl@bbb09ca[MainActivity](20744): ViewPostIme pointer 1
D/InputMethodManager(20744): prepareNavigationBarInfo() DecorView@f37afbe[MainActivity]
D/InputMethodManager(20744): getNavigationBarColor() -855310
V/InputMethodManager(20744): Starting input: tba=com.example.dev_flutter ic=null mNaviBarColor -855310 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false
D/InputMethodManager(20744): startInputInner - Id : 0
I/InputMethodManager(20744): startInputInner - mService.startInputOrWindowGainedFocus
D/InputTransport(20744): Input channel destroyed: 'ClientS', fd=83
D/ViewRootImpl@bbb09ca[MainActivity](20744): stopped(true) old=false
D/SurfaceView(20744): windowStopped(true) false io.flutter.embedding.android.FlutterSurfaceView{d4a9e58 V.E...... ........ 0,0-1080,2301} of ViewRootImpl@bbb09ca[MainActivity]
D/SurfaceView(20744): surfaceDestroyed callback.size 1 #1 io.flutter.embedding.android.FlutterSurfaceView{d4a9e58 V.E...... ........ 0,0-1080,2301}
D/SurfaceView(20744): remove() io.flutter.embedding.android.FlutterSurfaceView{d4a9e58 V.E...... ........ 0,0-1080,2301} Surface(name=SurfaceView - com.example.dev_flutter/com.example.dev_flutter.MainActivity@d4a9e58@0)/@0xda2e070
D/ViewRootImpl@bbb09ca[MainActivity](20744): MSG_WINDOW_FOCUS_CHANGED 0 1
D/InputMethodManager(20744): prepareNavigationBarInfo() DecorView@f37afbe[MainActivity]
D/InputMethodManager(20744): getNavigationBarColor() -855310
E/ViewRootImpl@bbb09ca[MainActivity](20744): Surface is not valid.

``` 

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel dev, 1.21.0-1.0.pre, on Mac OS X 10.15.6 19G73, locale en-GB)
    • Flutter version 1.21.0-1.0.pre at /Users/taha/Code/flutter_dev
    • Framework revision f25bd9c55c (7 days ago), 2020-07-14 20:26:01 -0400
    • Engine revision 99c2b3a245
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 20bf2fcf56)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (5 available)
    • SM M305F (mobile)      • 32003c30dc19668f          • android-arm64  • Android 10 (API 29)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS 13.6
    • macOS (desktop)        • macos                     • darwin-x64     • Mac OS X 10.15.6 19G73
    • Web Server (web)       • web-server                • web-javascript • Flutter Tools
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome 84.0.4147.89

• No issues found!
```

</details>
